tests/py/unit/test_build_steps.py

```
@@ -0,0 +1,1078 @@
+        workspace = tmp_path / "workspace"
+        entry_point = tmp_path / "main.tsx"
+        entry_point.write_text("// entry")
+
+        run_order: list[str] = []
+
+        class Step1(BuildStep):
+            @property
+            def name(self) -> str:
+                return "step1"
+
+            def run(self, ctx: BuildContext) -> None:
```

Ruff (0.14.13)

Unused method argument: `ctx`

(ARG002)